experiments/yarn/src/main/scala/org/apache/gearpump/experiments/yarn/client/Client.scala

```diff
@@ -162,44 +168,118 @@ class Client(configuration:AppConfig, yarnConf: YarnConfiguration, yarnClient: Y
     })
   }
 
-  def monitorAM(appContext: ApplicationSubmissionContext): Unit = {
-    val appId = appContext.getApplicationId
+  def delay(appId: ApplicationId): (ApplicationReport, YarnApplicationState) = {
+    Thread.sleep(1000)
+    val appReport = yarnClient.getApplicationReport(appId)
+    val appState = appReport.getYarnApplicationState
+    (appReport, appState)
+  }
+
+  def monitor(appId: ApplicationId): Unit = {
```

Comment:
You may want to script additional actions to be completed after the application exits - so it would have value in these cases where particular clean could be done after appmaster exists. This could be done using curl and parsing the application's state from a GET but would be much more involved.